### PR TITLE
Refactor max_connections to max_concurrency

### LINKED ISSUE #7507: [Storage] Non breaking Changes fromAPI preview
- `BlobBlock` shouldn't have block_id as optional
-  mypy error in abort_copy
- mypy error in append_block

### LINKED ISSUE #7512: Storage - Rename max_connections to max_concurrency

### PR DESCRIPTION
**Pull request by rakshith91**
...and some other minor changes
- `BlobBlock` shouldn't have block_id as optional
- mypy error in abort_copy
- mypy error in append_block

Resolves #7512 
Resolves #7507

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-blob/azure/storage/blob/_shared/downloads.py [hunk 1]
```diff
@@ -383,32 +383,32 @@ def _initial_request(self):
 
         return response
 
-    def content_as_bytes(self, max_connections=1):
+    def content_as_bytes(self, max_concurrency=1):
```
Comment (annatisch) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_shared/downloads.py]: I guess moving this parameter to the constructor will be a separate PR specifically for the download stream API?

File: sdk/storage/azure-storage-blob/azure/storage/blob/models.py [hunk 2]
```diff
@@ -768,7 +768,7 @@ class BlobBlock(DictMixin):
         Block size in bytes.
     """
 
-    def __init__(self, block_id=None, state=BlockState.Latest):
+    def __init__(self, block_id, state=BlockState.Latest):
```
Comment (annatisch) [inline sdk/storage/azure-storage-blob/azure/storage/blob/models.py]: If you change this you have to update the `from_generated` classmethod on line 777 below to pass in the required param

File: sdk/storage/azure-storage-blob/azure/storage/blob/_shared/downloads.py (hunk 1 above)
Comment (rakshith91) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_shared/downloads.py]: yes, I'll do it here https://github.com/Azure/azure-sdk-for-python/issues/7504

**Comment by rakshith91** [comment]
It seems like the test failure is not related to rehydrate request..merging this
https://dev.azure.com/azure-sdk/public/_build/results?buildId=121935&view=ms.vss-test-web.build-test-results-tab&runId=4822112&paneView=debug&resultId=100550
